powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/service/RegistrationService.java

```diff
@@ -0,0 +1,116 @@
+import com.wultra.security.powerauth.client.model.response.fido2.RegistrationChallengeResponse;
+import com.wultra.security.powerauth.client.model.response.fido2.RegistrationResponse;
+import com.wultra.security.powerauth.fido2.configuration.WebAuthnConfiguration;
+import com.wultra.security.powerauth.fido2.controller.request.RegisterCredentialRequest;
+import com.wultra.security.powerauth.fido2.controller.request.RegistrationOptionsRequest;
+import com.wultra.security.powerauth.fido2.controller.response.RegistrationOptionsResponse;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Service for WebAuthn registration tasks.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@Service
+@AllArgsConstructor
+@Slf4j
+public class RegistrationService {
+
+    private final PowerAuthFido2Client fido2Client;
+    private final SharedService sharedService;
+    private final WebAuthnConfiguration webauthnConfig;
+
+    /**
+     * Build public key registration options.
+     * @param request Request form with user input.
+     * @return Public key registration options.
+     * @throws PowerAuthClientException in case of PowerAuth server communication error.
+     */
+    public RegistrationOptionsResponse registerOptions(final RegistrationOptionsRequest request) throws PowerAuthClientException {
+        final String userId = request.username();
+        final String applicationId = request.applicationId();
+
+        final RegistrationChallengeResponse challengeResponse = fetchChallenge(userId, applicationId);
+
+        logger.info("Building registration options for userId={}, applicationId={}", userId, applicationId);
+        return RegistrationOptionsResponse.builder()
+                .rp(new PublicKeyCredentialRpEntity(webauthnConfig.getRpId(), webauthnConfig.getRpName()))
+                .user(new PublicKeyCredentialUserEntity(userId.getBytes(), userId, userId))
+                .challenge(challengeResponse.getChallenge())
+                .pubKeyCredParams(List.of(
+                        new PublicKeyCredentialParameters(PublicKeyCredentialType.PUBLIC_KEY, COSEAlgorithmIdentifier.ES256)
+                ))
+                .timeout(webauthnConfig.getTimeout().toMillis())
+                .excludeCredentials(sharedService.fetchExistingCredentials(userId, applicationId))
+                .build();
+    }
+
+    /**
+     * Register credential at PowerAuth server.
+     * @param credential Newly created public key credential
+     * @return PowerAuth registration response.
+     * @throws PowerAuthClientException in case of PowerAuth server communication error.
+     */
+    public RegistrationResponse register(final RegisterCredentialRequest credential) throws PowerAuthClientException {
+        logger.info("Registering created credential of userId={}, applicationId={}", credential.activationName(), credential.applicationId());
```

Comment:
The `userId` parameter is printed as `credential.activationName()`, is this correct?